Fixes to issues #49 #50 and #51

rcolodzin opened the pull request.

colodzin committed:
Fixes to issues #49 #50 and #51

jillh510 commented on lambda/marchon.py:
```diff
@@ -63,9 +63,11 @@ def get_event_data():
     fields = {'name': 0, 'eventDate': 1, 'eventLink': 2, 'location': 3, 'host': 4,
               'affiliate': 5, 'contactName': 6, 'contactEmail': 7, 'facebook': 8,
-              'twitter': 9, 'instagram': 10, 'motpLink': 12}
+              'twitter': 9, 'instagram': 10, 'motpLink': 12, 'eventLink': 12}
```
eventLink is in column 2 of the spreadsheet and is already read in.

rcolodzin replied:
Will roll back this change

jillh510 commented on map.css:
```diff
@@ -6,7 +6,7 @@ body {
 
 body.events {
-  background-color: #ed135c;
+  /* background-color: #ed135c; */
```
This file is used by both index.html and events.html, and the change should only be made for events.html.

rcolodzin replied:
In index.html, the body tag does not have the 'events' class. So, this change would apply to events.html, family_separation_events.html, marchonpolls_events.html, but not to index.html. What do you think, should we keep it?

jillh510 commented on index.html:
```diff
@@ -25,8 +25,8 @@
               <div class="info pull-left">
-                <div v-if="activeGroup.motpLink" class="event-link">
-                  <a :href="activeGroup.motpLink" target="_blank">Find out more</a>
+                <div v-if="activeGroup.eventLink" class="event-link">
```
Looks like March On's data still has some events from last year in their spreadsheet. I've asked for them to be removed, but until they are, let's not take this change.

rcolodzin replied:
What if we were to put in code that handled both cases, so that it will work with whichever label is passed? This of course assumes they are never BOTH passed...
Like this:
`                 <div v-if="activeGroup.eventLink" class="event-link">
                  <a :href="activeGroup.eventLink" target="_blank">Find out more</a>
                </div>
                <div v-if="activeGroup.motpLink" class="event-link">
                  <a :href="activeGroup.motpLink" target="_blank">Find out more</a>
                </div>
`

jillh510 replied:
Good point. It could use v-else-if to avoid the chance of seeing both.

jillh510 commented on map.js:
```diff
@@ -171,67 +177,81 @@ const app = new Vue({
         }
         // familySepEvents
         if (familySepEventsPast.features.length) {
+  		  var icon = (isEventsPage ? defaultIcon : 'star-gray-light');
```
Family separation events shouldn't wind up on the events page, so no need for this.

jillh510 commented on map.js:
```diff
@@ -171,67 +177,81 @@ const app = new Vue({
           });
         }
         if (familySepEventsFuture.features.length) {
+    	  var icon = (isEventsPage ? defaultIcon : 'star-blue');
```
Family separation events shouldn't wind up on the events page, so no need for this

jillh510 commented on map.js:
```diff
@@ -171,67 +177,81 @@ const app = new Vue({
         }
 
         if (marchonpollsFlagshipEvents.features.length) {
+      	  var icon = (isEventsPage ? defaultIcon : 'star-pink');
```
Poll events shouldn't wind up on the events page, so no need for this

jillh510 commented on map.js:
```diff
@@ -171,67 +177,81 @@ const app = new Vue({
           });
         }
         if (marchonpollsEvents.features.length) {
+      	  var icon = (isEventsPage ? defaultIcon : 'star-black');
```
Poll events shouldn't wind up on the events page, so no need for this

colodzin committed:
Eventlink was already there, this change was redundant and likely could create other problems.

colodzin committed:
Only check if it is the events page for affiliateTrue in map.js
In index.html, check for motpLink and for eventLink, as some data has the older name, and some the newer one.

colodzin committed:
change the check for motpLink to v-else-if, so we only get there if there is no eventLink property

jillh510 committed:
do not use eventLink until we sort out handling old events
